src/test/java/hw2/ex1/LoginSeleniumTestWithSoftAssert.java

```diff
@@ -0,0 +1,80 @@
+        List<String> expectedText = Arrays.asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+        List<String> actualText = elements.stream().map(WebElement::getText).collect(Collectors.toList());
+        softAssert.assertEquals(actualText, expectedText, "Menu buttons have not proper texts");
+
+        //6. Assert that there are 4 images on the Index Page and they are displayed
+        softAssert.assertEquals(getWait().until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.className("benefit-icon"))).size(),
+                4, "The amount of images is incorrect");
+
+        //7. Assert that there are 4 texts on the Index Page under icons and they have proper text
+        List<WebElement> benefitsText = getWait().until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.className("benefit-txt")));
+        softAssert.assertEquals(benefitsText.size(), 4);
+        List<String> expectedBenefitTexts = Arrays.asList(
+                "To include good practices\n" + "and ideas from successful\n" + "EPAM project\n",
+                "To be flexible and\n" + "customizable\n",
+                "To be multiplatform\n",
+                "Already have good base\n" + "(about 20 internal and\n" + "some external projects),\n" + "wish to get more…");
+        List<String> actualBenefitTexts = benefitsText.stream().map(WebElement::getText).collect(Collectors.toList());
+        softAssert.assertEquals(actualBenefitTexts, expectedBenefitTexts, "Texts under icons are incorrect");
+
+        //8. Assert that there is the iframe with “Frame Button” exist
+        softAssert.assertTrue(getWait().until(ExpectedConditions.visibilityOfElementLocated(
+                By.id("frame"))).isDisplayed(), "The iframe is not exists");
+
+        //9. Switch to the iframe and check that there is “Frame Button” in the iframe
+        getDriver().switchTo().frame("frame");
+        softAssert.assertTrue(getWait().until(ExpectedConditions.visibilityOfElementLocated(
+                By.id("button-frame"))).isSelected(), "The “Frame Button” is not exists");
+
+        //10. Switch to original window back
+        getDriver().switchTo().defaultContent();
+
+        //11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+        List<WebElement> leftSection = getWait().until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.id("mCSB_1")));
+        softAssert.assertEquals(leftSection.size(), 5);
+        String expectedLeftSectionTexts = "Home\n" +
+                "Contact form\n" +
+                "Service\n" +
+                "Metals & Colors\n" +
+                "Elements packs";
+        String actualLeftSectionTexts = getWait().until(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.id("mCSB_1"))).toString();
+        softAssert.assertEquals(actualLeftSectionTexts, expectedLeftSectionTexts, "Left section menu items have incorrect proper text");
```

Comment:
You have missed one important thing for the softAssertion. Now your test will have only one result **PASSED**

Reply:
added assertAll() method and fixed bugs due to this